File: android/src/main/java/org/maplibre/reactnative/components/layers/types/fillextrusion/MLRNFillExtrusionLayerManager.java
package org.maplibre.reactnative.components.layers.types.fillextrusion;

import com.facebook.react.uimanager.ThemedReactContext;

import org.maplibre.reactnative.components.layers.MLRNVectorLayerManager;

public class MLRNFillExtrusionLayerManager extends MLRNVectorLayerManager<MLRNFillExtrusionLayer> {
    public static final String REACT_CLASS = "MLRNFillExtrusionLayer";

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected MLRNFillExtrusionLayer createLayerViewInstance(ThemedReactContext reactContext) {
        return new MLRNFillExtrusionLayer(reactContext);
    }
}

File: android/src/main/java/org/maplibre/reactnative/components/layers/types/fillextrusion/MLRNFillExtrusionLayerManager.kt
package org.maplibre.reactnative.components.layers.types.fillextrusion

import com.facebook.react.uimanager.ThemedReactContext
import org.maplibre.reactnative.components.layers.MLRNVectorLayerManager

class MLRNFillExtrusionLayerManager : MLRNVectorLayerManager<MLRNFillExtrusionLayer?>() {
    override fun getName(): String {
        return REACT_CLASS
    }

    override fun createLayerViewInstance(reactContext: ThemedReactContext?): MLRNFillExtrusionLayer {
        return MLRNFillExtrusionLayer(reactContext)
    }

    companion object {
        const val REACT_CLASS: String = "MLRNFillExtrusionLayer"
    }
}
